packages/transactions/src/useDraftDocument.ts

```diff
@@ -0,0 +1,121 @@
+import { useEffect } from 'react';
+import {
+  UseStore,
+  useMutation,
+  useQuery,
+  zustand,
+} from '@openmsupply-client/common';
+
+interface DraftStore<T> {
+  draft: T | null;
+  setDraft: (draft: T | null) => void;
+}
+
+/**
+ * Create a store for a draft record. This can be used for each type
+ * of entity, such that we can create a separate store for each type of entity,
+ * so that if we are creating two types of entities at the same time,
+ * we ensure they're separated. For example, if we are creating a location
+ * while creating a new invoice.
+ */
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const createDraftStore = <T>(): UseStore<DraftStore<T>> => {
+  return zustand<DraftStore<T>>(set => ({
+    draft: null,
+    setDraft: draft => set(state => ({ ...state, draft })),
+  }));
+};
```

Comment:
So as mentioned, this is for replacing what was just a `useState` call in the `useDraftDocument` hook. 

This means whenever `useDraftDocument` is used, it returns the same `draft` and also is hooked up to the same `api` calls through react-querys cache